bioptim/optimization/parameters.py

```diff
@@ -61,8 +63,20 @@ def __init__(
 
         super(Parameter, self).__init__(type=Parameters, **params)
         self.function = function
-        self.initial_guess = initial_guess
-        self.bounds = bounds
+
+        if not isinstance(scaling, np.ndarray):
+            scaling = np.asarray(scaling, dtype=float)
+        if len(scaling.shape) == 1:
+            self.scaling = scaling[:, np.newaxis]
+        elif len(scaling.shape) == 2:
+            if scaling.shape[1] != 1:
+                raise RuntimeError(
+                    f"Invalid ncols for Parameter Scaling "
+                    f"(ncols = {scaling.shape[1]}), the expected number of column is 1"
+                )
+
+        self.initial_guess = initial_guess.scale(self.scaling)
```

Comment:
Please remove the return value, and copy it first and then scale. I think what you have do can be considered good practice, but is not in accordance with anything else in the API. So for consistency I'd go the two lines way

Reply:
Ok, ignore my previous comments then